src/colmap/scene/database_cache.cc

```diff
@@ -71,9 +71,31 @@ std::shared_ptr<DatabaseCache> DatabaseCache::Create(
   timer.Restart();
   LOG(INFO) << "Loading matches...";
 
+  std::unordered_set<image_t> image_ids;
+  std::vector<class Image> images;
+
+  {
+    images = database.ReadAllImages();
+
+    // Determines for which images data should be loaded.
+    if (image_names.empty()) {
+      for (const auto& image : images) {
+        image_ids.insert(image.ImageId());
+      }
+    } else {
+      for (const auto& image : images) {
+        if (image_names.count(image.Name()) > 0) {
+          image_ids.insert(image.ImageId());
+        }
+      }
+    }
+  }
+
   std::vector<image_pair_t> image_pair_ids;
   std::vector<TwoViewGeometry> two_view_geometries;
-  database.ReadTwoViewGeometries(&image_pair_ids, &two_view_geometries);
+  database.ReadTwoViewGeometries(&image_pair_ids,
```

Comment:
Does this really save much runtime or memory? Below in L130, we are already applying that same filter?

Reply:
Sorry for the late reply. This PR filters two_view_geometry based on image_ids, which can reduce the memory occupied by two_view_geometries. In particular, when the user only processes a small subset of all images, such as hierarchical SfM based on incremental SfM, each computing node only reconstructs a small part of all images.